pick.lic

```diff
@@ -2,523 +2,680 @@
+      echo 'Dodged a trap with Safe...'
+      trapped = false
+    when *@disarm_succeeded
+      trapped = false
     end
+
+    box['trapped'] = trapped
+    box['trap_difficulty'] += 1 if Flags['disarm-shift']
+
+    if Flags['more-traps']
+      echo "Saw another trap after disarming/tripping one..." if @debug
+      box['trapped'] = true
+      box['trap_difficulty'] = nil
+    end
+
+    echo "Updated box after disarm attempt: #{box}" if @debug
   end
 
-  def unlock_box_with_key?(box)
-    return false unless @settings.use_skeleton_key
-    DRCI.get_item_if_not_held?(@settings.skeleton_key)
-    result = bput("turn my #{@settings.skeleton_key} at my #{box}", "^You turn", "that doesn't seem to do much", "I could not find", "What were you referring")
-    waitrt?
-    return result =~ /^You turn/
+  def handle_trap_sprung
+    DRC.message('**SPRUNG TRAP**')
+    DRC.message('**SPRUNG TRAP**')
+
+    pause 1 while stunned?
+
+    health = DRCH.check_health
+    echo "health: bleeders - #{health['bleeders']} : poison: #{health['poisoned']} - score: #{health['score']}" if @debug
+    return unless health['bleeders'].any? || health['poisoned'] || health['score'] > @settings.saferoom_health_threshold
+
+    echo "Things are bad enough to go heal..." if @debug
+    snapshot = Room.current.id
+    DRC.wait_for_script_to_complete('safe-room')
+
+    echo "handle_trap_sprung: safe-room finished - returning to room #{snapshot}" if @debug
+    DRCT.walk_to(snapshot)
+
+    echo "handle_trap_sprung: danger handled" if @debug
   end
 
-  def attempt_pick(box)
-    find_lockpick unless @settings.use_lockpick_ring
-    pause 0.5 while pick(box)
+  def handle_trap_too_hard_or_blacklisted(box)
+    if @skipped_box_storage
+      return if DRCI.put_away_item?(box['noun'], @skipped_box_storage)
+      echo "Throwing away box cause it was not disarmed and can't be stowed" if @debug
+    else
+      echo "Throwing away box cause it was not disarmed and picking_box_storage is not set" if @debug
+    end
+
+    DRCI.dispose_trash(box['noun'], @worn_trashcan, @worn_trashcan_verb)
   end
 
-  def pick(box)
-    waitrt?
-    check_danger
-    case bput("pick my #{box} ident", @pick_careful, @pick_quick, @pick_blind, @pick_retry, /Find a more appropriate tool and try again/, /It's not even locked, why bother/)
-    when /Find a more appropriate tool and try again/
-      find_lockpick
-      pick(box)
+  def identify_lock(box)
+    echo "identify_lock(#{box})" if @debug
+
+    pick_identify_match = DRC.bput("pick my #{box['noun']} ident",
+      @trap_sprung_matches,
+      @pick_identify_messages,
+      @pick_retry,
+      /Find a more appropriate tool and try again/,
+      /It's not even locked, why bother/)
+
+    case pick_identify_match
+    when *@trap_sprung_matches
+      handle_trap_sprung
+    when /Find a more appropriate tool and try again/, *@pick_retry
+      box['locked'] = true
+      box['lock_difficulty'] = nil
```

Comment:
Need to flag not to use lockpick ring so will fall back to trying to get a loose lockpick, and if that fails then script will naturally exit with warning message to user. Otherwise you get into an infinite loop identifying the box. Aligns with logic at around line 522 in the `pick(box)` method.
```suggestion
      box['lock_difficulty'] = nil
      @use_lockpick_ring = false
```